### PR TITLE
Skip tests on CI when Subversion is not installed locally

### PR DESCRIPTION
**Pull request by jglick**
Amends #194 which was merged prematurely. Subsumes #193. Quick fix to get builds passing. Better would be to improve tests to run portably, as mentioned in comments.

@reviewbybees

### CODE REVIEW

File: src/test/java/hudson/scm/AbstractSubversionTest.java [hunk 1]
```diff
@@ -57,12 +67,9 @@ public static Proc runSvnServe(URL zip) throws Exception {
      * Runs svnserve to serve the specified directory as a subversion repository.
      */
     public static Proc runSvnServe(File repo) throws Exception {
```
Comment (jglick) [inline src/test/java/hudson/scm/AbstractSubversionTest.java]: Probably this utility could be better moved to `SubversionSampleRepoRule` as well, but leaving for another day.

**Comment by jglick** [comment]
[Latest test failures](https://ci.jenkins.io/job/Plugins/job/subversion-plugin/job/PR-195/4/testReport/) seem to be random

```
java.net.SocketException: Permission denied: listen
```

File: pom.xml [hunk 2]
```diff
@@ -121,16 +121,20 @@ THE SOFTWARE.
...
-      <version>4.11</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
       <groupId>org.mockito</groupId>
       <artifactId>mockito-core</artifactId>
       <version>1.9.5</version>
       <scope>test</scope>
+      <exclusions>
+        <exclusion>
```
Comment (oleg-nenashev) [inline pom.xml]: It makes sense to mention conflicted versions in comments when you fix upper bounds

File: pom.xml [hunk 3]
```diff
@@ -143,6 +147,12 @@ THE SOFTWARE.
       <artifactId>powermock-api-mockito</artifactId>
       <version>1.5</version>
       <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>org.mockito</groupId>
+          <artifactId>mockito-all</artifactId>
```
Comment (oleg-nenashev) [inline pom.xml]: Ideally this dependency should be removed at all from the plugin. Mockito Core should be enough
https://github.com/mockito/mockito/wiki/Declaring-mockito-dependency

File: src/test/java/hudson/scm/AbstractSubversionTest.java [hunk 4]
```diff
@@ -46,23 +53,41 @@ protected void configureSvnWorkspaceFormat(int format) throws Exception {
...
-    	this.descriptor.configure(req, formData);
+    	r.jenkins.getDescriptorByType(SubversionSCM.DescriptorImpl.class).configure(req, formData);
+    }
+
+    public static void checkForSvnServe() throws InterruptedException {
+        LocalLauncher launcher = new LocalLauncher(StreamTaskListener.fromStdout());
+        try {
+            launcher.launch().cmds("svnserve","--help").start().join();
+        } catch (IOException e) {
+            // TODO better to add a docker-fixtures test dep so CI builds can run these tests
```
Comment (oleg-nenashev) [inline src/test/java/hudson/scm/AbstractSubversionTest.java]: +1, fine for now

File: src/test/java/hudson/scm/SVNWorkingCopyTest.java [hunk 5]
```diff
@@ -23,57 +23,56 @@
...
-import org.tmatesoft.svn.core.wc.SVNStatusClient;
-import org.tmatesoft.svn.core.wc2.SvnGetStatus;
-
-import java.io.File;
-import org.junit.Ignore;
 
 /**
  * @author schristou88
  */
 @Ignore("TODO org.tmatesoft.svn.core.SVNException: svn: E175002: PROPFIND of '/trunk/hudson/test-projects/trivial-ant': 405 Method Not Allowed (https://svn.jenkins-ci.org)")
```
Comment (oleg-nenashev) [inline src/test/java/hudson/scm/SVNWorkingCopyTest.java]: Do we still need to ignore it? CC @christ66

**Comment by jglick** [comment]
> Maybe the flaky test should be just disabled

It is not a specific test. On that build, several tests failed, including `InjectedTest`, and not because of test code.

File: pom.xml (hunk 2 above)
Comment (jglick) [inline pom.xml]: The `objenesis` conflict is with PowerMock. This is pretty standard—every plugin using PowerMock winds up with the same conflict. Ditto `hamcrest-core`, conflicting with a newer one from `junit` / `jenkins-test-harness`.

File: pom.xml (hunk 3 above)
Comment (jglick) [inline pom.xml]: It is possible a newer version of `powermock-api-mockito` fixes their dependencies; I did not want to upgrade PowerMock here.

Anyone publishing `*-all` artifacts on Central with `jar` packaging should be given a stern lecture…

File: src/test/java/hudson/scm/SVNWorkingCopyTest.java (hunk 5 above)
Comment (jglick) [inline src/test/java/hudson/scm/SVNWorkingCopyTest.java]: We do: `svn.jenkins-ci.org` is long dead. Another case where we need to use `docker-fixtures` to set up a test server.

(There is another test floating around here that connects to `subversion.tigris.org`, which still works, but is poor style.)

File: pom.xml (hunk 3 above)
Comment (jglick) [inline pom.xml]: To be clear, adding this `exclusion` _does_ remove this dependency from the plugin.

Comment (oleg-nenashev) [inline pom.xml]: OK, let's keep it as is for now

Comment (oleg-nenashev) [inline pom.xml]: Race condition in comments

**Pull request merged**